src/software/ai/hl/stp/play/indirect_free_kick_play.cpp

```diff
@@ -0,0 +1,293 @@
+     *   find a pass in time
+     * - Two robots try to get in good positions in the enemy end to receive a pass
+     * - Two robots crease defend
+     * - One robot is goalie
+     */
+
+    // Setup the goalie
+    auto goalie_tactic = std::make_shared<GoalieTactic>(
+        world.ball(), world.field(), world.friendlyTeam(), world.enemyTeam());
+
+    // Setup crease defenders to help the goalie
+    std::array<std::shared_ptr<CreaseDefenderTactic>, 2> crease_defender_tactics = {
+        std::make_shared<CreaseDefenderTactic>(world.field(), world.ball(),
+                                               world.friendlyTeam(), world.enemyTeam(),
+                                               CreaseDefenderTactic::LeftOrRight::LEFT),
+        std::make_shared<CreaseDefenderTactic>(world.field(), world.ball(),
+                                               world.friendlyTeam(), world.enemyTeam(),
+                                               CreaseDefenderTactic::LeftOrRight::RIGHT),
+    };
+
+    // If the passing is coming from the friendly end, we split the cherry-pickers
+    // across the x-axis in the enemy half
+    Rectangle cherry_pick_1_target_region(world.field().centerPoint(),
+                                          world.field().enemyCornerPos());
+    Rectangle cherry_pick_2_target_region(world.field().centerPoint(),
+                                          world.field().enemyCornerNeg());
+
+    // Otherwise, the pass is coming from the enemy end, put the two cherry-pickers
+    // on the opposite side of the x-axis to wherever the pass is coming from
+    if (world.ball().position().x() > -1)
+    {
+        double y_offset =
+            -std::copysign(world.field().xLength() / 2, world.ball().position().y());
+        cherry_pick_1_target_region =
+            Rectangle(Point(0, world.field().xLength() / 4),
+                      Point(world.field().xLength() / 2, y_offset));
+        cherry_pick_2_target_region =
+            Rectangle(Point(0, world.field().xLength() / 4), Point(0, y_offset));
+    }
+    // These two tactics will set robots to roam around the field, trying to put
+    // themselves into a good position to receive a pass
+    auto cherry_pick_tactic_1 =
+        std::make_shared<CherryPickTactic>(world, cherry_pick_1_target_region);
+    auto cherry_pick_tactic_2 =
+        std::make_shared<CherryPickTactic>(world, cherry_pick_2_target_region);
+
+    // This tactic will move a robot into position to initially take the free-kick
+    auto align_to_ball_tactic = std::make_shared<MoveTactic>();
+    align_to_ball_tactic->addWhitelistedAvoidArea(AvoidArea::BALL);
+
+    PassGenerator pass_generator(world, world.ball().position(),
+                                 PassType::RECEIVE_AND_DRIBBLE);
+    pass_generator.setTargetRegion(
+        Rectangle(Point(0, world.field().yLength() / 2), world.field().enemyCornerNeg()));
```

Comment:
why that corner?

Reply:
This is actually the entire enemy half of the field, not just a corner. Maybe we want member function to get the middle sideline points of the field? Not sure the right name, but can make a ticket if we think this is something we use relatively often (which I think we do).

Reply:
maybe it's worth adding `getPositiveXHalf()` and `getNegativeXHalf()` functions to the `Field` class?

Reply:
created #979